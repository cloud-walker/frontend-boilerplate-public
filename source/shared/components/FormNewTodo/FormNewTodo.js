import React, {PropTypes} from 'react'
import {compose, withState, withProps} from 'recompose'

/**
* Form to create new todos.
*/

const enhance = compose(
  withState('value', 'setValue', ''),
  withProps(({value, setValue, addTodo}) => ({
    handleSubmit: e => (
      e.preventDefault(),
      addTodo(value),
      setValue('')
    ),
    handleChange: e => setValue(e.target.value),
  }))
)

const Component = ({value, handleSubmit, handleChange}) =>
  <form onSubmit={handleSubmit}>
    <input
      type="text"
      value={value}
      onChange={handleChange}
    />
    <input type="submit" value="add"/>
  </form>

Component.displayName = 'FormNewTodo'
Component.propTypes = {
  value: PropTypes.string.isRequired,
  handleSubmit: PropTypes.func.isRequired,
  handleChange: PropTypes.func.isRequired,
}

export default enhance(Component)
