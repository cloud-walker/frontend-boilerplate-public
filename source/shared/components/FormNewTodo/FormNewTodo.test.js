import React from 'react'
import {shallow} from 'enzyme'
import FormNewTodo from './index'

test('it should render properly', () => {
  const wrapper = shallow(<FormNewTodo value="prova"/>)

  expect(wrapper).toMatchSnapshot()
})
