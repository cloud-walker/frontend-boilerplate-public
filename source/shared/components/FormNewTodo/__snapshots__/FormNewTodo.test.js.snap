exports[`test it should render properly 1`] = `
<FormNewTodo
  handleChange={[Function]}
  handleSubmit={[Function]}
  setValue={[Function]}
  value="" />
`;
